client/dashboard/store-performance/index.js

```diff
@@ -16,99 +22,196 @@ import {
+					newPrefs = prefs;
+				} else {
+					newPrefs = prefs.filter( pref => pref !== statKey );
+				}
+				this.props.updateCurrentUserData( {
+					dashboard_performance_indicators: newPrefs,
+				} );
+				return {
+					userPrefs: newPrefs,
+				};
+			} );
 		};
 	}
 
 	renderMenu() {
+		const { indicators } = this.props;
 		return (
 			<EllipsisMenu label={ __( 'Choose which analytics to display', 'wc-admin' ) }>
 				<MenuTitle>{ __( 'Display Stats:', 'wc-admin' ) }</MenuTitle>
-				<MenuItem onInvoke={ this.toggle( 'showCustomers' ) }>
-					<ToggleControl
-						label={ __( 'Show Customers', 'wc-admin' ) }
-						checked={ this.state.showCustomers }
-						onChange={ this.toggle( 'showCustomers' ) }
-					/>
-				</MenuItem>
-				<MenuItem onInvoke={ this.toggle( 'showProducts' ) }>
-					<ToggleControl
-						label={ __( 'Show Products', 'wc-admin' ) }
-						checked={ this.state.showProducts }
-						onChange={ this.toggle( 'showProducts' ) }
-					/>
-				</MenuItem>
-				<MenuItem onInvoke={ this.toggle( 'showOrders' ) }>
-					<ToggleControl
-						label={ __( 'Show Orders', 'wc-admin' ) }
-						checked={ this.state.showOrders }
-						onChange={ this.toggle( 'showOrders' ) }
-					/>
-				</MenuItem>
+				{ indicators.map( ( indicator, i ) => {
+					const checked = ! this.state.userPrefs.includes( indicator.stat );
+					return (
+						<MenuItem onInvoke={ this.toggle( indicator.stat ) } key={ i }>
+							<ToggleControl
+								label={ sprintf( __( 'Show %s', 'wc-admin' ), indicator.label ) }
+								checked={ checked }
+								onChange={ this.toggle( indicator.stat ) }
+							/>
+						</MenuItem>
+					);
+				} ) }
 			</EllipsisMenu>
 		);
 	}
 
-	render() {
-		const totalOrders = 10;
-		const totalProducts = 1000;
-		const { showCustomers, showProducts, showOrders } = this.state;
+	renderList() {
+		const {
+			query,
+			primaryRequesting,
+			secondaryRequesting,
+			primaryError,
+			secondaryError,
+			primaryData,
+			secondaryData,
+			userIndicators,
+		} = this.props;
+		if ( primaryRequesting || secondaryRequesting ) {
+			return <SummaryListPlaceholder numberOfItems={ userIndicators.length } />;
+		}
+
+		if ( primaryError || secondaryError ) {
+			return null;
+		}
+
+		const persistedQuery = getPersistedQuery( query );
+
+		const { compare } = getDateParamsFromQuery( query );
+		const prevLabel =
+			'previous_period' === compare
+				? __( 'Previous Period:', 'wc-admin' )
+				: __( 'Previous Year:', 'wc-admin' );
+		return (
+			<SummaryList>
+				{ userIndicators.map( ( indicator, i ) => {
+					const primaryItem = find( primaryData.data, data => data.stat === indicator.stat );
+					const secondaryItem = find( secondaryData.data, data => data.stat === indicator.stat );
 
+					if ( ! primaryItem || ! secondaryItem ) {
+						return null;
+					}
+
+					const href =
+						( primaryItem._links &&
+							primaryItem._links.report[ 0 ] &&
+							primaryItem._links.report[ 0 ].href ) ||
+						'';
```

Comment:
Should this and `reportUrl` default to `false` so `SummaryNumber` knows there isn't a link if one was not supplied?